spec/reservation_tracker_spec.rb

```diff
@@ -0,0 +1,162 @@
+        expect( reservation ).must_be_instance_of(Hotel::Reservation)
+      end
+    end
+
+    it 'returns correct number of reservations' do
+      expect( @list_of_reservations.length ).must_equal(2)
+    end
+
+  end
+
+  describe 'available_rooms method' do
+
+    it 'returns empty array if no rooms available' do
+
+      [*1..20].each do |room|
+        @hotel.book_reservation(room, Date.new(2018,6,4), Date.new(2018,7,7))
+      end
+
+      expect(
+        @hotel.available_rooms(Date.new(2018,6,4), Date.new(2018,7,7))
+      ).must_equal([])
+      expect(
+        @hotel.available_rooms(Date.new(2018,6,1), Date.new(2018,6,5))
+      ).must_equal([])
+      expect(
+        @hotel.available_rooms(Date.new(2018,7,1), Date.new(2018,7,15))
+      ).must_equal([])
+    end
+
+    it 'returns list of available rooms' do
+
+      [*1..10].each do |room|
+        @hotel.book_reservation(room, Date.new(2018,6,4), Date.new(2018,7,7))
+      end
+
+      expect(
+        @hotel.available_rooms(Date.new(2018,6,4), Date.new(2018,7,7))
+      ).must_be_instance_of(Array)
+      expect(
+        @hotel.available_rooms(Date.new(2018,7,6), Date.new(2018,7,7))
+      ).must_equal([*11..20])
+
+
+      [*11..19].each do |room|
+        @hotel.book_reservation(room, Date.new(2018,6,4), Date.new(2018,7,7))
+      end
+
+      expect(
+        @hotel.available_rooms(Date.new(2018,6,10), Date.new(2018,7,7))
+      ).must_equal([20])
+    end
+  end
+
+  describe 'book_reservation' do
+    before do
+      @hotel.book_reservation(1, Date.new(2018,7,4), Date.new(2018,7,8))
+      @hotel.book_reservation(1, Date.new(2018,8,4), Date.new(2018,8,5))
+      @hotel.book_reservation(2, Date.new(2018,6,4), Date.new(2018,7,7))
+    end
+
+    it 'raises error if date is not in correct format' do
+      expect{
+        @hotel.book_reservation(1, '20180503', Date.new(2018,5,9))
+      }.must_raise(ArgumentError)
+      expect{
+        @hotel.book_reservation(1, Date.new(2018,5,9), '20180503')
+      }.must_raise(ArgumentError)
+    end
+
+    it 'raises error if room is unavailable' do
+      expect{
+        @hotel.book_reservation(1, Date.new(2018,7,7), Date.new(2018,7,10))
+      }.must_raise(StandardError)
+      expect{
+        @hotel.book_reservation(2, Date.new(2018,5,1), Date.new(2018,6,4))
+      }.must_raise(StandardError)
+    end
+
+    it 'successful book returns reservation with matching dates' do
+      reservation = @hotel.book_reservation(5, Date.new(2018,7,7), Date.new(2018,7,10))
+
+      expect( reservation ).must_be_instance_of(Hotel::Reservation)
+      expect( reservation.start_date ).must_equal(Date.new(2018,7,7))
+      expect( reservation.end_date ).must_equal(Date.new(2018,7,10))
+    end
+
+    it 'successful book increases total reservations array by one' do
+      initial_reservations = @hotel.reservations.length
+      reservation = @hotel.book_reservation(5, Date.new(2018,7,7), Date.new(2018,7,10))
```

Comment:
A small nitpicking thing-- the local variable `reservation` isn't used in this scope anywhere. If you think that's true too, it's probably okay to get rid of it?